spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ArmeriaSpringActuatorAutoConfiguration.java

```diff
@@ -179,13 +185,29 @@ ArmeriaServerConfigurator actuatorServerConfigurator(
             ManagementServerProperties serverProperties,
             BeanFactory beanFactory,
             ArmeriaSettings armeriaSettings) {
-        final EndpointMapping endpointMapping = new EndpointMapping(properties.getBasePath());
+        // default base-path is empty, so we can just concat it
+        final EndpointMapping endpointMapping =
+                new EndpointMapping(serverProperties.getBasePath() + properties.getBasePath());
         final Collection<ExposableWebEndpoint> endpoints = endpointsSupplier.getEndpoints();
         return sb -> {
             final Integer managementPort = obtainManagementServerPort(sb, beanFactory, serverProperties);
             if (managementPort != null) {
                 addLocalManagementPortPropertyAlias(environment, managementPort);
             }
+
+            if (serverProperties.getSsl() != null &&
+                (serverProperties.getSsl().getKeyStore() != null ||
+                 serverProperties.getSsl().getTrustStore() != null)) {
+                logger.warn("Use armeria.ssl instead of management.server.ssl to set key or trust store, " +
+                            "management.server.ssl will be ignored except management.server.ssl.enabled");
+            }
+
+            if (serverProperties.getSsl() != null && serverProperties.getSsl().isEnabled() &&
+                armeriaSettings.getSsl() != null) {
+                // TODO how to detect duplication better?
+                // configureTls(sb, armeriaSettings.getSsl());
```

Comment:
How to detect if we set tls already, setting twice will throw "TLS has been configured already." exception.
Maybe we can use InternalServices bean exists or not, but like webflux auto configuration,  we may have early return at https://github.com/line/armeria/blob/ddd39fa5f4c566d764199bd3adb2a8aa1f704db6/spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java#L133 and skip `configureServerWithArmeriaSettings`.

Reply:
> How to detect if we set tls already, 

Because it's set already, can we just ignore the TSL setting here and let `configureServerWithArmeriaSettings` handle it?

Reply:
Oops, I realized that we can set the TLS setting to the `VirtualHostBuilder`. 😅 
How about creating [`VirtualHostBuilder`](https://github.com/line/armeria/blob/91c787f0a087499453103480f0fc8ab01072f9ab/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1416 ) and setting the TLS settings to the builder?

Reply:
So if `management.server.address` can use `InetAddress#getHostName` and use it for `hostnamePattern`?

Reply:
> can use InetAddress#getHostName and use it for hostnamePattern?

I believe so. It will also cover the case where only the IP address is specified.

So there're three cases for creating `VirtualHost`:
- When only the port is specified
  - We can create a virtual host using the port: https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1395
- When only the address is specfied
  - We can create a virtual host using the InetAddress#getHostName: https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1365
- When the port and address are specified.
  - - We can create a virtual host using the combination of the address and port (e.g. `localhost:8080`) https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1365

Reply:
Just double confirm, if I do
```
   sb.virtualHost(8081)
                   // tls related setting
                  .tls() 
                  ....
```
this still uses the ServerBuilder's key, am I correct? I tried it like https://github.com/line/armeria/issues/4560#issuecomment-1340503323 and looks like the key just uses host name pattern?

Reply:
> this still uses the ServerBuilder's ssl setting. 

Do you mean it uses ServerBuilder's ssl setting even though we set a new one?

Reply:
IIRC https://github.com/line/armeria/pull/4563/files#diff-5c2c0d3bb86e9e00a6565c64a590c2b9cfd2f009d9fa19b7d8d81fe4f8c3b4beR124 this won't work even I set ssl on port based virtualhost.

Reply:
One question about this, the address here may not the host pattern of ssl. So we can't use the address like it, right?

Reply:
So how do we setup the TLS settings when `ssl.enabled` is `true`? 🤔

Reply:
We can just use ServerBuilder's certificate setting I think. So `ssl.enabled` just enable the https protocol for that address/port?

Reply:
Had a chat with @kojilin and we have decided to just enable https protocol when `ssl.enabled` is `true`.